conman/routes/managers.py

```diff
@@ -30,3 +30,17 @@ def best_match_for_path(self, path):
         except IndexError:
             msg = 'No matching Route for URL. (Have you made a root Route?)'
             raise self.model.DoesNotExist(msg)
+
+    def move_branch(self, old_url, new_url):
+        """
+        Move a Route and all descendants to a new URL.
+
+        Movement will be atomic. This means that failure to move any one Route
+        will cause all movement to fail. A conflicting URL will cause an
+        IntegrityError.
+        """
+        to_move = self.filter(url__startswith=old_url)
+        to_move.update(url=Concat(
+            Value(new_url),
+            Substr('url', len(old_url) + 1),  # 1 indexed
```

Comment:
Thanks for noticing! This is one of my favourite parts of the whole codebase `:D`